src/wp-includes/blocks.php

```diff
@@ -238,6 +238,7 @@ function register_block_type_from_metadata( $file_or_folder, $args = array() ) {
 		'styles'          => 'styles',
 		'example'         => 'example',
 		'apiVersion'      => 'api_version',
+		'variations'      => 'variations',
```

Comment:
Note: Some of the properties that `variations` have are translatable, so it would need more work if we decide to go this way.

Reply:
There is quite basic support for i18n in `block.json`:
https://github.com/WordPress/wordpress-develop/blob/4a09e7d87767b768589b940c8ca0f5fbea98cfd1/src/wp-includes/blocks.php#L249-L275

It's interconnected with WP-CLI as implemented here: https://github.com/wp-cli/i18n-command/pull/210

There is also the documentation for `block.json` that needs an update - https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-metadata.md.

I think it should be rather extracted to its own patch because of the complexity. With that, for start, we would allow to dynamically register `variations` on the server and it would be successfully propagated to the client.

Reply:
> I think it should be rather extracted to its own patch because of the complexity.

Sounds reasonable 👍 I'm happy to follow up on that if this one lands.

Reply:
@gziolo Dropped in eae37b9 ✅ Let me know if I missed any other spots.